### PR TITLE
drivers: wifi: wifi_winc1500: Updated driver capabilities.

### PR DESCRIPTION
**Pull request by Kludentwo**
Added ap_enable api. The driver will open create a access point
with DHCP Server ip 192.168.1.1 and no security.

Added a small fix for the AF_INET issue.

Added parent and remote to accept routine context.

Signed-off-by: Nicolai Glud <nicolai.glud@prevas.dk>

### CODE REVIEW

File: drivers/wifi/winc1500/wifi_winc1500.c [hunk 1]
```diff
@@ -890,17 +899,33 @@ static void handle_socket_msg_accept(struct socket_data *sd, void *pvMsg)
...
+		/** Setup remote */
+		a_sd->context->remote.sa_family = AF_INET;
+		net_sin(&a_sd->context->remote)->sin_port =
+			accept_msg->strAddr.sin_port;
+		net_sin(&a_sd->context->remote)->sin_addr.s_addr =
+			accept_msg->strAddr.sin_addr.s_addr;
+		a_sd->context->flags |= NET_CONTEXT_REMOTE_ADDR_SET;
+
+		sd->accept_cb(a_sd->context,
 				      (struct sockaddr *)&accept_msg->strAddr,
```
Comment (jukkar) [inline drivers/wifi/winc1500/wifi_winc1500.c:924]: Can you fix the indentation here for the accept_cb() params.

File: drivers/wifi/winc1500/wifi_winc1500.c [hunk 2]
```diff
@@ -305,7 +305,11 @@ static int winc1500_get(sa_family_t family,
 		return -1;
```
Comment (jukkar) [inline drivers/wifi/winc1500/wifi_winc1500.c:305]: In the commit subject, could you write `drivers: wifi: winc1500: .....` instead, as there is no need to mention wifi in the driver name.

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by Kludentwo** [comment]
Changes since Jukkar's comments:
Added ap_disable. 
Fixed 2 warnings with respect to close function.
Fixed indentation.

**Comment by jukkar** [comment]
@tbursztyka can you check this one, it has been pending long time.

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**